lib/main_common.pm

```diff
@@ -655,8 +655,16 @@ sub boot_hdd_image {
     if (get_var('UEFI') && (get_var('BOOTFROM') || get_var('BOOT_HDD_IMAGE'))) {
         loadtest "boot/uefi_bootmenu";
     }
-    loadtest "support_server/wait_support_server" if get_var('USE_SUPPORT_SERVER');
-    loadtest "boot/boot_to_desktop";
+    # TODO why does HA_CLUSTER need on non-aarch64 the same schedule?
+    if (check_var('ARCH', 'aarch64') || get_var('HA_CLUSTER')) {
+        # We need to boot *before* waiting for support server
+        # Because of TianoCore UEFI boot method
+        loadtest "boot/boot_to_desktop";
+        loadtest "support_server/wait_support_server" if get_var('USE_SUPPORT_SERVER');
+    } else {
+        loadtest "support_server/wait_support_server" if get_var('USE_SUPPORT_SERVER');
+        loadtest "boot/boot_to_desktop";
+    }
```

Comment:
@asmorodskyi @ldevulder please check this section as you know more about why HPC and HA cluster tests should differ in the order of loading boot_to_desktop and wait_support_server

Reply:
@okurz  the reason (was discussed several times already) is network availability after boot_to_deskop. If systems boots without support server => no dhcp during boot, it sometimes will not get lease in time after wait for support server and there is very high chance, that next test (qam patch_and_reboot) will fail because of no network

Reply:
The swap between boot and wait and following failure was discussed in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/4020

Reply:
I think at the end we need to find a way to properly boot in the same way on x86_64 BIOS/UEFI and aarch64 UEFI.

But I don't have enough time now to do this change...